at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -46,75 +50,112 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
-          onError(syncResult);
-        }
-        return syncResult;
+        _logger.severe(
+            'skipping sync - ${syncRequest._id} since due to network unavailability');
+        continue;
       }
-      var serverCommitId;
       try {
-        // Check if local and cloud secondary are in sync. If true, return.
         if (await isInSync()) {
-          syncResult = SyncResult();
-          _logger.info('Local Secondary and Cloud Secondary are in sync');
-          // Setting isSyncInProgress to false, to allow next sync call.
-          _isSyncInProgress = false;
-          if (onDone != null) {
-            onDone(syncResult);
-          }
-          return syncResult;
+          _logger.finer('server and local are in sync - ${syncRequest._id}');
+          syncRequest.result!
+            ..syncStatus = SyncStatus.success
+            ..lastSyncedOn = DateTime.now().toUtc();
+          continue;
         }
-        // Get latest server commit id.
-        serverCommitId = await _getServerCommitId();
-      } on AtLookUpException catch (exception) {
+      } on AtLookUpException catch (e) {
         _logger.severe(
-            '${_atClient.getCurrentAtSign()} ${exception.errorMessage}');
-        syncResult = SyncResult();
-        syncResult.syncStatus = SyncStatus.failure;
-        syncResult.atClientException =
-            AtClientException(exception.errorCode, exception.errorMessage);
-        // Setting isSyncInProgress to false, to allow next sync call.
-        _isSyncInProgress = false;
-        if (onError != null) {
-          onError(syncResult);
-        }
-        return syncResult;
+            'Atlookup exception for sync ${syncRequest._id}. Reason ${e.toString()}');
+        syncRequest.result!.atClientException =
+            AtClientException(e.errorCode, e.errorMessage);
+        continue;
+      }
+      final serverCommitId = await _getServerCommitId();
+      await _sync(serverCommitId, syncRequest);
+    }
+    _clearQueue();
+  }
+
+  void _onDone() {
+    //#TODO implement
+  }
+
+  void _onError() {
+    //#TODO implement
+  }
+
+  void _clearQueue() {
+    _syncRequests
+        .removeWhere((syncRequest) => _checkSyncRequestStatus(syncRequest));
+  }
+
+  bool _checkSyncRequestStatus(SyncRequest syncRequest) {
+    if (syncRequest.result != null &&
+        syncRequest.result!.syncStatus == SyncStatus.success) {
+      if (syncRequest.onDone != null) {
+        syncRequest.onDone!(syncRequest.result);
       }
-      // Sync
-      return await _sync(serverCommitId, onDone, onError);
-    } finally {
-      _isSyncInProgress = false;
+      return true;
+    } else if (syncRequest.result != null &&
+            syncRequest.result!.syncStatus == SyncStatus.failure ||
+        syncRequest.requestedOn.difference(DateTime.now().toUtc()).inMinutes >
+            _syncMaxRetryTime) {
+      if (syncRequest.onError != null) {
+        syncRequest.onError!(syncRequest.result);
```

Comment:
I think onError need not be called. Will explain in a call.

Reply:
done